src/OneWayBondingCurve.sol

```diff
@@ -60,23 +55,32 @@ contract OneWayBondingCurve {
 
     /// @notice Purchase USDC for BAL
     /// @param amountIn Amount of BAL input
+    /// @param withdrawFromAave Whether to receive as USDC (true) or aUSDC (false)
     /// @return amountOut Amount of USDC received
     /// @dev Purchaser has to approve BAL transfer before calling this function
-    function purchase(uint256 amountIn) external returns (uint256) {
+    function purchase(uint256 amountIn, bool withdrawFromAave) external returns (uint256) {
         if (amountIn == 0) revert OnlyNonZeroAmount();
         if (amountIn > availableBalToBeFilled()) revert ExcessBalAmountIn();
 
         uint256 amountOut = getAmountOut(amountIn);
         if (amountOut == 0) revert OnlyNonZeroAmount();
 
         totalBalReceived += amountIn;
-        totalUsdcPurchased += amountOut;
+        totalAusdcPurchased += amountOut;
 
         // Execute the purchase
         BAL.safeTransferFrom(msg.sender, AaveV2Ethereum.COLLECTOR, amountIn);
-        USDC.safeTransferFrom(AaveV2Ethereum.COLLECTOR, msg.sender, amountOut);
+        if (withdrawFromAave) {
+            AUSDC.safeTransferFrom(AaveV2Ethereum.COLLECTOR, address(this), amountOut);
+            // Compensating for +1/-1 precision issues due to rounding on aTokens while it's being transferred
+            amountOut = amountOut - 1;
```

Comment:
this is a bit unnecessary imprecision and will leave dust on this contract. Instead you can do
```
uint256 preciseAmountOut = AaveV2Ethereum.POOL.withdraw(address(USDC), type(uint256).max, msg.sender);
emit Purchase(address(BAL), address(USDC), amountIn, preciseAmountOut);
```

Reply:
What if someone had accidentally sent aUSDC directly to the bonding curve contract? Wouldn't the above move all that aUSDC to whoever called purchase at the time?

Reply:
That was the main reason I didn't go with the above approach. Let me know if you feel otherwise.

Reply:
Done and fixed.